repos/spack_repo/builtin/packages/precice/package.py

```diff
@@ -125,6 +125,8 @@ class Precice(CMakePackage):
     conflicts("%clang@:3.7")
     conflicts("%intel@:16")
 
+    patch("edge-incomplete-type.patch", when="@2.0.0")
```

Comment:
The patch was [first released in v2.1.0](https://github.com/precice/precice/commit/a0d05f4e7f030837ca98181f786067bc793deae9), but there have also been v2.0.1 and v2.0.2 in the meantime, which are also affected. How about:

```suggestion
    patch("edge-incomplete-type.patch", when="@2.0")
```

This [should cover all v2.0.x](https://spack.readthedocs.io/en/latest/packaging_guide_creation.html#specifying-version-constraints).